File: decompiled/sources/com/hjq/widget/view/FloatActionButton.java
package com.hjq.widget.view;

import android.animation.ValueAnimator;
import android.content.Context;
import android.util.AttributeSet;
import androidx.appcompat.widget.AppCompatImageView;

public final class FloatActionButton extends AppCompatImageView {
    private static final int ANIM_TIME = 300;
    private final Runnable mHideRunnable = new Runnable() {
        public final void run() {
            FloatActionButton.this.lambda$new$3$FloatActionButton();
        }
    };
    private final Runnable mShowRunnable = new Runnable() {
        public final void run() {
            FloatActionButton.this.lambda$new$1$FloatActionButton();
        }
    };

    public FloatActionButton(Context context) {
        super(context);
    }

    public FloatActionButton(Context context, AttributeSet attributeSet) {
        super(context, attributeSet);
    }

    public FloatActionButton(Context context, AttributeSet attributeSet, int i) {
        super(context, attributeSet, i);
    }

    public void show() {
        removeCallbacks(this.mHideRunnable);
        postDelayed(this.mShowRunnable, 600);
    }

    public void hide() {
        removeCallbacks(this.mShowRunnable);
        post(this.mHideRunnable);
    }

    public /* synthetic */ void lambda$new$1$FloatActionButton() {
        if (getVisibility() == 4) {
            setVisibility(0);
        }
        ValueAnimator ofFloat = ValueAnimator.ofFloat(new float[]{0.0f, 1.0f});
        ofFloat.setDuration(300);
        ofFloat.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
            public final void onAnimationUpdate(ValueAnimator valueAnimator) {
                FloatActionButton.this.lambda$null$0$FloatActionButton(valueAnimator);
            }
        });
        ofFloat.start();
    }

    public /* synthetic */ void lambda$null$0$FloatActionButton(ValueAnimator valueAnimator) {
        float floatValue = ((Float) valueAnimator.getAnimatedValue()).floatValue();
        setAlpha(floatValue);
        setScaleX(floatValue);
        setScaleY(floatValue);
    }

    public /* synthetic */ void lambda$new$3$FloatActionButton() {
        if (getVisibility() != 4) {
            ValueAnimator ofFloat = ValueAnimator.ofFloat(new float[]{1.0f, 0.0f});
            ofFloat.setDuration(300);
            ofFloat.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
                public final void onAnimationUpdate(ValueAnimator valueAnimator) {
                    FloatActionButton.this.lambda$null$2$FloatActionButton(valueAnimator);
                }
            });
            ofFloat.start();
        }
    }

    public /* synthetic */ void lambda$null$2$FloatActionButton(ValueAnimator valueAnimator) {
        float floatValue = ((Float) valueAnimator.getAnimatedValue()).floatValue();
        setAlpha(floatValue);
        setScaleX(floatValue);
        setScaleY(floatValue);
        if (floatValue == 0.0f) {
            setVisibility(4);
        }
    }
}

File: decompiled/sources/com/hjq/widget/view/FloatActionButton.kt
package com.hjq.widget.view

import android.animation.ValueAnimator
import android.content.Context
import android.util.AttributeSet
import android.view.View
import androidx.appcompat.widget.AppCompatImageView

class FloatActionButton : AppCompatImageView {
    private val mHideRunnable = Runnable { hideNow() }
    private val mShowRunnable = Runnable { showNow() }

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    )

    fun show() {
        removeCallbacks(mHideRunnable)
        postDelayed(mShowRunnable, 600)
    }

    fun hide() {
        removeCallbacks(mShowRunnable)
        post(mHideRunnable)
    }

    private fun showNow() {
        if (visibility == View.INVISIBLE) {
            visibility = View.VISIBLE
        }
        val animator = ValueAnimator.ofFloat(0f, 1f)
        animator.duration = ANIM_TIME
        animator.addUpdateListener { animation ->
            val value = animation.animatedValue as Float
            alpha = value
            scaleX = value
            scaleY = value
        }
        animator.start()
    }

    private fun hideNow() {
        if (visibility != View.INVISIBLE) {
            val animator = ValueAnimator.ofFloat(1f, 0f)
            animator.duration = ANIM_TIME
            animator.addUpdateListener { animation ->
                val value = animation.animatedValue as Float
                alpha = value
                scaleX = value
                scaleY = value
                if (value == 0f) {
                    visibility = View.INVISIBLE
                }
            }
            animator.start()
        }
    }

    companion object {
        private const val ANIM_TIME = 300L
    }
}
